coap/api/transport.go

```diff
@@ -166,10 +169,10 @@ func decodeMessage(msg *mux.Message) (*messaging.Message, error) {
 }
 
 func parseKey(msg *mux.Message) (string, error) {
-	if obs, _ := msg.Options.Observe(); obs != 0 && msg.Code == codes.GET {
+	if obs, _ := msg.Observe(); obs != 0 && msg.Code() == codes.GET {
 		return "", nil
 	}
-	authKey, err := msg.Options.GetString(message.URIQuery)
+	authKey, err := msg.Options().GetString(message.URIQuery)
```

Comment:
@1998-felix When publishing with `coap-client` , it sends cancel `obs` after 30 seconds.To cancel `observation` , value should > 0 , mostly it will be 1 .

```bash
coap-client  -p 5683   coap://localhost/channels/$CHANNEL_ID/messages -O 15,auth=$THING_KEY  -O 6,0x00  -T Token1 -v 7
```
If we get cancel obs request from client , we simply unsubscribe the event 

But now we will get 4.01 unauthorized error.

Because of this line, we get 4.01 error during unsubscribe. Since `obs` will `1`, This line returns empty string, so it fails in authorization.

we can remove this check, because this is preventing to return actual key.

```suggestion

```